crescent/ext/cooldowns/__init__.py

```diff
@@ -0,0 +1,35 @@
+from typing import Awaitable, Callable, Optional
+
+from hikari import Snowflakeish
+from pycooldown import FixedCooldown
+
+from crescent import Context, HookResult
+
+CooldownCallbackT = Callable[[Context, float], Awaitable[Optional[HookResult]]]
+
+
+def cooldown(
+    period: float, capacity: int, callback: Optional[CooldownCallbackT] = None
+) -> Callable[[Context], Awaitable[Optional[HookResult]]]:
+    """
+    Ratelimit implementation using a sliding window.
+
+    period:
+        The amount of seconds between each ratelimit.
```

Comment:
A little wordy. Maybe "The period of time, in seconds, between cooldown resets"?